### PR TITLE
V7 update

### PR DESCRIPTION
**Pull request by ayobuenavista**

### CODE REVIEW

File: .solhint.json [hunk 1]
```diff
@@ -9,9 +9,15 @@
     "avoid-throw": "error",
     "avoid-tx-origin": "off",
     "check-send-result": "error",
-    "compiler-version": ["error","0.6.6"],
+    "compiler-version": ["error", "0.7.6"],
     "const-name-snakecase": "error",
     "func-name-mixedcase": "error",
+    "func-visibility": [
```
Comment (ducquangkstn) [inline .solhint.json:15]: we can remove this by upgrade solhint

File: .gitignore [hunk 2]
```diff
@@ -48,3 +48,4 @@ report
 
 # Environment
 .env
+bin/
```
Comment (manhlx3006) [inline .gitignore]: missing eof

Comment (Anyhowclick) [inline .gitignore]: CRLF

**Review by manhlx3006 — DISMISSED** [review]

File: hardhat.config.js [hunk 3]
```diff
@@ -119,9 +114,14 @@ module.exports = {
...
 
-  solc: {
-    version: '0.6.6',
-    optimizer: require('./solcOptimiserSettings.js'),
+  solidity: {
+    version: '0.7.6',
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 430,
```
Comment (Anyhowclick) [inline hardhat.config.js]: try increasing the runs?

File: package.json [hunk 4]
```diff
@@ -28,25 +29,24 @@
   },
   "homepage": "https://github.com/KyberNetwork/kyber_utils_sc#readme",
   "devDependencies": {
-    "@nomiclabs/buidler": "1.4.5",
-    "@nomiclabs/buidler-truffle5": "1.3.4",
-    "@nomiclabs/buidler-web3": "1.3.4",
-    "@openzeppelin/test-helpers": "0.5.6",
-    "chai": "4.2.0",
+    "@nomiclabs/hardhat-truffle5": "^2.0.0",
```
Comment (Anyhowclick) [inline package.json]: remove `^`

**Review by Anyhowclick — CHANGES_REQUESTED** [review]

File: contracts/mock/ReentrancyAttack.sol [hunk 5]
```diff
@@ -1,4 +1,5 @@
-pragma solidity 0.6.6;
+// SPDX-License-Identifier: MIT
```
Comment (ducquangkstn) [inline contracts/mock/ReentrancyAttack.sol]: Pls remove Reentrancy test. 
We use open-zeppelin directly

File: package.json (hunk 4 above)
Comment (ducquangkstn) [inline package.json]: Pls remove ^

File: .solhint.json (hunk 1 above)
Comment (ayobuenavista) [reply .solhint.json:15]: Hmmm, I'm using latest solhint, but still receiving this message: `Explicitly mark visibility in function (Set ignoreConstructors to true if using solidity >=0.7.0)`

That's why I had to add the rule explicitely.

File: hardhat.config.js (hunk 3 above)
Comment (ayobuenavista) [reply hardhat.config.js]: 100,000 runs

File: .solhint.json (hunk 1 above)
Comment (manhlx3006) [reply .solhint.json:15]: for sol 7 we don't need the func visibility for constructor right?

**Review by manhlx3006 — APPROVED** [review]

**Review by Anyhowclick — APPROVED** [review]

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**